disco_aws_automation/disco_subnet.py

```diff
@@ -111,16 +127,53 @@ def recreate_route_table(self):
         self._associate_route_table(new_route_table)
         self._route_table = new_route_table
 
-    def create_nat_gateway(self, eip_allocation_id):
-        """ Create a NAT gateway for the subnet"""
-        self.nat_eip_allocation_id = eip_allocation_id
-        self._nat_gateway = self.nat_gateway
+    def create_nat_gateway(self, eip_allocation_id=None, use_dyno_nat=False):
```

Comment:
Could it be cleaner to have a single argument here. Default to dyno_nat if `eip_allocation_id` is None?